File: demo/misc/file_demo.py


#需要进行显示关闭
#f=file("/path_to_file/file_name","r")   #file() open() 两者类似返回值都是file类
###读操作
f=open("file_path","r")   ###打开文件
f.read()                  ###读文件 read()会一次性读取文件的全部内容  可以反复调用read(size)方法读取定长内容
f.close()				  ###关闭文件

###写操作
f = open('/Users/michael/test.txt', 'w')    ###打开文件
f.write('Hello, world!')					###写操作
f.close()									###关闭文件


#结束模块时自动关闭 不必显式关闭
with open("/path_to_file/file_name","r") as f:
    f.read()

try:
    f = open('/path/to/file', 'r')
    print f.read()
finally:
    if f:
        f.close()


#移动文件指针
#当调用read时，指针会移动到文件尾部，再次read前需要先移动文件指针
#或者以读写方式打开，写文件后想读取，则需要先移动文件指针
f.seek(0)  

"""
'r', 'w','a'  读 写 增加
'b'  二进制形式打开
'+'  允许同时读写 

全部模式
r rb r+ rb+ 
w wb w+ wb+
a ab a+ ab+
"""

File: others/lua/file_demo.lua
-- 以只读方式打开文件
file = io.open("/tmp/test.txt", "r")
io.input(file)
print(io.read())
io.close(file)


-- 以附加的方式打开只写文件
file = io.open("/tmp/test.txt", "a")
io.output(file)
io.write("hello\n")
io.close(file)

--[[
io.read                                                    
"*n"	读取一个数字并返回它。                                file.read("*n")
"*a"	从当前位置读取整个文件。                              file.read("*a")
"*l"    默认，读取下一行，在文件尾 (EOF) 处返回 nil。         file.read("*l")
number	返回一个指定字符个数的字符串，或在 EOF 时返回 nil。   file.read(5)
--]]


--[[
io.open 模式
r	以只读方式打开文件，该文件必须存在。
w	打开只写文件，若文件存在则文件长度清为0，即该文件内容会消失。若文件不存在则建立该文件。
a	以附加的方式打开只写文件。若文件不存在，则会建立该文件，如果文件存在，写入的数据会被加到文件尾，即文件原先的内容会被保留。（EOF符保留）
r+	以可读写方式打开文件，该文件必须存在。
w+	打开可读写文件，若文件存在则文件长度清为零，即该文件内容会消失。若文件不存在则建立该文件。
a+	与a类似，但此文件可读可写
b	二进制模式，如果文件是二进制文件，可以加上b
+	号表示对文件既可以读也可以写
--]]


-- 以只读方式打开文件
file = io.open("/tmp/test.txt", "r")
print(file:read())
file:close()

-- 以附加的方式打开只写文件
file = io.open("/tmp/test.txt", "a")
file:write("hello\n")
--file:flush()
file:close()

